ffs/filcold/filcold.go

```diff
@@ -158,56 +157,67 @@ func (fc *FilCold) renewDeal(ctx context.Context, c cid.Cid, size uint64, p ffs.
+		return ffs.FilStorage{}, failedStartedDeals, fmt.Errorf("starting renew deal proposal: %s", err)
+	}
+	successDeals, failedExecutingDeals, err := fc.WaitForDeals(ctx, c, okDeals)
 	if err != nil {
-		return ffs.FilStorage{}, errors, fmt.Errorf("executing renewed deal: %s", err)
+		return ffs.FilStorage{}, failedExecutingDeals, fmt.Errorf("waiting for renweal deal: %s", err)
 	}
-	if len(props) != 1 {
-		return ffs.FilStorage{}, errors, fmt.Errorf("unsuccessful renewal")
+	if len(successDeals) != 1 {
+		fc.l.Log(ctx, c, "Renewal deal execution errored: %s", failedStartedDeals[0].Message)
+		return ffs.FilStorage{}, failedExecutingDeals, fmt.Errorf("unsuccessful renewal")
 	}
-	return props[0], errors, nil
+	return successDeals[0], nil, nil
 }
 
-func (fc *FilCold) makeDeals(ctx context.Context, c cid.Cid, size uint64, cfgs []deals.StorageDealConfig, fcfg ffs.FilConfig) ([]ffs.FilStorage, []ffs.DealError, error) {
+// makeDeals starts deals with the specified miners. It returns a slice with all the ProposalCids
+// that were started successfully, and a slice of DealError with deals that failed to be started.
+func (fc *FilCold) makeDeals(ctx context.Context, c cid.Cid, size uint64, cfgs []deals.StorageDealConfig, fcfg ffs.FilConfig) ([]cid.Cid, []ffs.DealError, error) {
 	for _, cfg := range cfgs {
 		fc.l.Log(ctx, c, "Proposing deal to miner %s with %d fil per epoch...", cfg.Miner, cfg.EpochPrice)
 	}
 
-	var sres []deals.StoreResult
-	sres, err := fc.dm.Store(ctx, fcfg.Addr, c, size, cfgs, uint64(fcfg.DealDuration))
+	sres, err := fc.dm.Store(ctx, fcfg.Addr, c, size, cfgs, uint64(fcfg.DealMinDuration))
 	if err != nil {
 		return nil, nil, fmt.Errorf("storing deals in deal module: %s", err)
 	}
-
-	proposals, errors, err := fc.waitForDeals(ctx, c, sres, fcfg.DealDuration)
-	if err != nil {
-		return nil, errors, fmt.Errorf("waiting for deals to finish: %s", err)
+	var okDeals []cid.Cid
+	var failedDeals []ffs.DealError
+	for _, r := range sres {
+		if !r.Success {
+			fc.l.Log(ctx, c, "Proposal with miner %s failed: %s", r.Config.Miner, r.Message)
+			log.Warnf("failed store result: %s", r.Message)
+			de := ffs.DealError{
+				ProposalCid: r.ProposalCid,
+				Message:     r.Message,
+				Miner:       r.Config.Miner,
+			}
+			failedDeals = append(failedDeals, de)
+			continue
+		}
+		okDeals = append(okDeals, r.ProposalCid)
 	}
-	return proposals, errors, nil
+	return okDeals, failedDeals, nil
 }
 
-func (fc *FilCold) waitForDeals(ctx context.Context, c cid.Cid, storeResults []deals.StoreResult, duration int64) ([]ffs.FilStorage, []ffs.DealError, error) {
+// WaitForDeals blocks waiting for all the provided proposals to reach a final status.
+// It returns the ones that finished successfully and the other deal errors that happened on failed ones.
+func (fc *FilCold) WaitForDeals(ctx context.Context, c cid.Cid, proposals []cid.Cid) ([]ffs.FilStorage, []ffs.DealError, error) {
```

Comment:
Since now `Store` doesn't wait for deals, some of the work `waitForDeals` was doing is more clearly done by `Store`. Now `WaitForDeals` has a clearer logic.